### DISCUSSION TITLE
Combine and move

### DISCUSSION THREAD

**Discussion by istvan-b01** [spakin/SimpInkScr#92] [Q&A] [answered]
Hi,

Can someone please help with a probably simple task.
I generate a bunch (loads) of rectangles which I want to combine (union) into a single path, then do the same with another generated bunch of rectangles, then subtract the result of the second one from the first one and move the result into a layer.
so schematically something like

rect1 = rect(...).to_path()
rect2 = rect(...).to_path()
rect3 = rect(...).to_path()
rect4 = rect(...).to_path()
layer0=layer('name')
arr1 = [rect1, rect2]
arr2 = [ rect3, rect4]
u1=union(arr1)
u2=union(arr2)
diff=difference( u1, u2)
layer0.append(diff)

I managed to have partial results, like union, but the result does not seem to be  single object, and I cant move it into a layer therefore.
Been trying to do this for long, please help.

To put it in context, I'm trying to use Inkscape to create a font with loads of characters (for the  PUI area) that later I'd use in forge font, and as far as I know each character must be one layer that contains the single path of the glyph,
the glyphs are a bunch of non-overlapping rectangles, but ins some case they do and I need to extract some rectangles from other ones. The result needs be therefore on single path that I can move (within the script) into the corresponding layer.

So I feel that this is not something complicated I'm just not knowledgeable enough to know how everything works within Inkscape and Simple Inkscape Script. I'd very much appreciate as well if someone could direct me to some literature for both as I believe this is a very much needed extension, without exaggerating, the most important one by far as it allows you to generate work by scripting.
Thanks in advance for any help.

**Comment by istvan-b01** [comment]
to summarize: all I need is what do I need to do with the result of a path operation in order to append the result to a layer.
I just can't find an example that can help with this.

Thank you.

**Comment by spakin** [comment, marked as answer]
`apply_path_operation` returns a list of resulting objects.  In the case of `union` of `difference` that will be a single-element list so you simply have to take the first element.  Here's an example I hope you find useful:

```Python
r1 = rect((0, 0), (400, 300), fill='red').to_path()
r2 = rect((300, 200), (500, 400), fill='red').to_path()
outer_shape = apply_path_operation('union', [r1, r2])[0]

r3 = rect((100, 50), (250, 200), fill='blue').to_path()
r4 = rect((50, 150), (150, 250), fill='blue').to_path()
inner_shape = apply_path_operation('union', [r3, r4])[0]

shape_with_hole = apply_path_operation('difference', [outer_shape, inner_shape])[0]

my_layer = layer('My new layer')
my_layer.append(shape_with_hole)
```

**Reply by istvan-b01** [reply]
thank you very much!!!, I figured it out myself as well only half an hour ago and came back to delete the post as it was just me not understanding what's happening until I started thinking rather than just trying things out :)
hence I had the opposite of what I needed like the result hidden, while the 'hole' stayed. that's when I realized the difference between shapes and paths :)

style(stroke_width=1, fill='rgb(0,0,0)')
r1 = rect((0, 0), (378, 756)).to_path()
r2 = rect((126, 252), (504, 1008)).to_path()
r0 = rect((189, 420), (315,588)).to_path()
arr = [r1]
arr.append(r2)
combo1 = apply_path_operation('union', [p for p in arr])
combo2 = apply_path_operation('difference', [combo1[0], r0])
l0 = layer('name', combo2[0])

anyway, absolutely awesome project!!! should come with Inkscape by default, being so essential!
big cheers!!!
